shap/cext/gpu_treeshap.h

```diff
@@ -849,7 +850,7 @@ void GetBinSegments(const PathVectorT& paths, const SizeVectorT& bin_map,
 
 struct DeduplicateKeyTransformOp {
   template <typename SplitConditionT>
-  __device__ thrust::pair<size_t, int64_t> operator()(
+  __host__ __device__ thrust::pair<size_t, int64_t> operator()(
```

Comment:
For my own understanding, what is the motivation behind adding the `__host__` decorator here, and also the new `#include <thrust/sort.h>`?

Reply:
The new include because there's a use of the `thrust::sort` function. The newer CUDA version has improved header dependency. Hence, one needs to include a specific header for each feature in thrust.

> what is the motivation behind adding the __host__ decorator here

I think the newer CUDA version requires this to deduce return type for device lambda function.